Browser_IDE/executionEnvironment_Internal.js

```diff
@@ -215,32 +215,35 @@ ReferenceError: test is not defined
 // Currently those are the only two forms supported, but this should account for the majority well enough.
 // It also doesn't parse the url style ones - it only needs to work for the local user's code, so good enough.
 
+
+
 function parseErrorStack(err){
     const stackParse = /(?:@|\()((?:[^;:`]|[:;`](?=.*(?:\/|\.)))*)`?;?:([0-9]*)/g;
     let stack = [...err.stack.matchAll(stackParse)];
 
     let stackIndex = 0;
 
-    //Should we limit this to only SplashKitArgumentError? (i.e if (err instanceof SplashKitArgumentError))
-
     // Unwind stack until we find user code:
     while(stackIndex < stack.length && !stack[stackIndex][1].startsWith(userCodeBlockIdentifier))
         stackIndex += 1;
 
-    if (stackIndex >= stack.length)
-        stackIndex = 0;
-
-    let lineNumber = stack[stackIndex][2];
+    // Only include the stack frames that are relevant to your code
+    stack = stack.slice(stackIndex, stackIndex + 2);
```

Comment:
Right now this is hard-coding it to report two lines from the stack - but what if there are less, or more? Make sure to output _all_ the messages relevant to the user. You can re-use the `while(stackIndex...` logic above to search for the last part of the stack in user code, and then output everything between.